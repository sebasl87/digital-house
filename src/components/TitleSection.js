import React from "react";
import { View, Text } from "react-native";

import { styles } from "../styles";

const TitleSection = ({ text }) => {
  return (
    <View
      style={{ marginVertical: 20, alignItems: "flex-start", width: "100%" }}
    >
      <Text style={styles.titleSection}>{text}</Text>
    </View>
  );
};

export default TitleSection;
