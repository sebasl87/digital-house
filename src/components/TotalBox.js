import React from "react";
import { View, Text } from "react-native";

import { styles } from "../../styles";

const TotalBox = ({ points }) => {
  return (
    <View
      style={{
        alignItems: "center",
        width: "100%",
      }}
    >
      <View style={styles.blueBox}>
        <Text style={styles.monthBlueBox}>Diciembre</Text>
        <View style={styles.totalContainerBox}>
          <Text style={styles.totalBlueBox}>
            {points && points.toString()}.00 pts
          </Text>
        </View>
      </View>
    </View>
  );
};

export default TotalBox;
