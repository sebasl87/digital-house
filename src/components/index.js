export { default as TitleSection } from "./TitleSection";
export { default as TotalBox } from "./TotalBox";
