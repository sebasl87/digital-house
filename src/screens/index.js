export { default as HomeScreen } from "./HomeScreen";
export { default as ProductDetail } from "./ProductDetail";
