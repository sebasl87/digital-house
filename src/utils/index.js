import { MONTHS } from "../constants";

export const getTotal = (arr) => {
  const sum = arr
    .filter((mov) => mov.is_redemption)
    .reduce(function (prev, current) {
      return prev + +current.points;
    }, 0);
  const rest = arr
    .filter((mov) => mov.is_redemption === false)
    .reduce(function (prev, current) {
      return prev + +current.points;
    }, 0);
  const result = (sum - rest).toLocaleString("en");
  return result;
};

export const formatDate = (fecha) => {
  const dateFormat =
    fecha.slice(8, 10) +
    " de " +
    MONTHS[fecha.slice(5, 7)] +
    ", " +
    fecha.slice(0, 4);

  return dateFormat;
};
