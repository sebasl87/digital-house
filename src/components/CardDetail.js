import React from "react";
import { View, Text } from "react-native";

import { styles } from "../../styles";
import { formatDate } from "../utils";
import TitleSection from "./TitleSection";

const MovTable = ({ product, points, isRedemption, date }) => {
  return (
    <View style={styles.containerProductInfo}>
      <TitleSection text="Detalles del producto" />
      <Text style={[styles.monthBlueBox, { color: "#000" }]}>
        Comprado el {formatDate(date)}
      </Text>
      <TitleSection
        style={{ marginBottom: 32 }}
        text={
          isRedemption
            ? "Con esta compra utilizaste"
            : "Con esta compra acumulaste"
        }
      />
      <Text
        style={[
          styles.monthBlueBox,
          { color: "#000", fontSize: 24, lineHeight: 33 },
        ]}
      >
        {points} puntos
      </Text>
    </View>
  );
};

export default MovTable;
