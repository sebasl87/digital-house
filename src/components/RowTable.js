import { useNavigation } from "@react-navigation/native";
import React from "react";
import { View, Text, Image, TouchableOpacity } from "react-native";

import { styles } from "../../styles";
import { formatDate } from "../utils";

const RowTable = ({ product, points, isRedemption, date, image }) => {
  const navigation = useNavigation();

  return (
    <TouchableOpacity
      onPress={() => {
        navigation.navigate("Product", {
          product,
          points,
          isRedemption,
          date,
          image,
        });
      }}
    >
      <View style={styles.rowTable}>
        <Image
          source={{ uri: image }}
          style={{ width: 55, height: 55, borderRadius: 10 }}
        />
        <View style={{ marginLeft: 12, width: 200 }}>
          <Text
            style={[styles.titleSection, { color: "#000", marginVertical: 5 }]}
          >
            {product}
          </Text>
          <Text style={[styles.dateText, { marginTop: 2 }]}>
            {formatDate(date)}
          </Text>
        </View>
        <View
          style={{
            alignItems: "center",
            flexDirection: "row",
          }}
        >
          {isRedemption ? (
            <Text style={[styles.monthBlueBox, { color: "#FF0000" }]}>-</Text>
          ) : (
            <Text style={[styles.monthBlueBox, { color: "#00B833" }]}>+</Text>
          )}
          <Text style={[styles.monthBlueBox, { color: "#000" }]}>{points}</Text>
          <View style={{ justifyContent: "center", alignItems: "flex-end" }}>
            <Text
              style={[styles.monthBlueBox, { color: "#000", marginLeft: 8 }]}
            >
              &gt;
            </Text>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );};

export default RowTable;
