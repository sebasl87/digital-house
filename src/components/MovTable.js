import React from "react";
import { View } from "react-native";

import { styles } from "../../styles";
import RowTable from "./RowTable";

const MovTable = ({ movements }) => {
  return (
    <View style={styles.table}>
      {movements &&
        movements.map(({ product, createdAt, points, is_redemption }) => (
          <RowTable
            date={createdAt}
            isRedemption={is_redemption}
            product={product}
            points={points}
          />
        ))}
    </View>
  );
};

export default MovTable;
