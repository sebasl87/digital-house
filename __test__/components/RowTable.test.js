import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { RowTable } from "../../src/components";
import { render, fireEvent } from "react-native-testing-library";

const mockHistoryPush = jest.fn();

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: () => ({
    navigate: mockHistoryPush,
  }),
}));

describe("RowTable component", () => {
  it("should render successfully with mock movement", () => {
    const rowTable = renderer
      .create(
        <RowTable
          date="2022-12-09T06:34:25.607Z"
          product="Handmade Metal Shoes"
          points={16434}
          image="https://loremflickr.com/640/480/transport"
          isRedemption={false}
        />
      )
      .toJSON();
    expect(rowTable).toMatchSnapshot();
  });

  it("should render successfully with mock movement and isRedemption", () => {
    const rowTable = renderer
      .create(
        <RowTable
          date="2022-12-09T06:34:25.607Z"
          product="Handmade Metal Shoes"
          points={16434}
          image="https://loremflickr.com/640/480/transport"
          isRedemption
        />
      )
      .toJSON();
    expect(rowTable).toMatchSnapshot();
  });

  it("should be redirect on press", () => {
    const { getByTestId } = render(
      <RowTable
        date="2022-12-09T06:34:25.607Z"
        product="Handmade Metal Shoes"
        points={16434}
        image="https://loremflickr.com/640/480/transport"
        isRedemption={false}
      />
    );
    fireEvent.press(getByTestId("touchableProduct"));

    expect(mockHistoryPush).toHaveBeenCalled();
  });
});
