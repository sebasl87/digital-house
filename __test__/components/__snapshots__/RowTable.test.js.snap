// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowTable component should render successfully with mock movement 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="touchableProduct"
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://loremflickr.com/640/480/transport",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 55,
          "width": 55,
        }
      }
    />
    <View
      style={
        {
          "marginLeft": 12,
          "width": 200,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#9B9898",
              "fontFamily": "Avenir",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 19,
            },
            {
              "color": "#000",
              "marginVertical": 5,
            },
          ]
        }
      >
        Handmade Metal Shoes
      </Text>
      <Text
        style={
          [
            {
              "color": "#000",
              "fontFamily": "Avenir",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
            },
            {
              "marginTop": 2,
            },
          ]
        }
      >
        09 de Diciembre, 2022
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 22,
            },
            {
              "color": "#00B833",
            },
          ]
        }
      >
        + 
      </Text>
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 22,
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        16434
      </Text>
      <View
        style={
          {
            "alignItems": "flex-end",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "800",
                "lineHeight": 22,
              },
              {
                "color": "#000",
                "marginLeft": 8,
              },
            ]
          }
        >
          &gt;
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`RowTable component should render successfully with mock movement and isRedemption 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="touchableProduct"
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://loremflickr.com/640/480/transport",
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 55,
          "width": 55,
        }
      }
    />
    <View
      style={
        {
          "marginLeft": 12,
          "width": 200,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#9B9898",
              "fontFamily": "Avenir",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 19,
            },
            {
              "color": "#000",
              "marginVertical": 5,
            },
          ]
        }
      >
        Handmade Metal Shoes
      </Text>
      <Text
        style={
          [
            {
              "color": "#000",
              "fontFamily": "Avenir",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 16,
            },
            {
              "marginTop": 2,
            },
          ]
        }
      >
        09 de Diciembre, 2022
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 22,
            },
            {
              "color": "#FF0000",
            },
          ]
        }
      >
        - 
      </Text>
      <Text
        style={
          [
            {
              "color": "#FFF",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "800",
              "lineHeight": 22,
            },
            {
              "color": "#000",
            },
          ]
        }
      >
        16434
      </Text>
      <View
        style={
          {
            "alignItems": "flex-end",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "800",
                "lineHeight": 22,
              },
              {
                "color": "#000",
                "marginLeft": 8,
              },
            ]
          }
        >
          &gt;
        </Text>
      </View>
    </View>
  </View>
</View>
`;
